design/src/docs/pokemon_service.md

```diff
@@ -0,0 +1,239 @@
+the business logic is written as protocol-agnostic and clients request a service by calling into them, similar to an RPC call.
+```rust
+aws_smithy_http_server::routing::Router::new_rest_json_router(vec![
+    {
+        let svc = crate::server_operation_handler_trait::operation(
+            registry.capture_pokemon_operation,
+        );
+```
+At this level, logging might be prohibited by the `@sensitive` trait. If there are no `@sensitive` shapes, the generated code looks like:
+```rust
+let request_fmt = aws_smithy_http_server::logging::sensitivity::RequestFmt::new();
+let response_fmt = aws_smithy_http_server::logging::sensitivity::ResponseFmt::new();
+let svc = aws_smithy_http_server::logging::InstrumentOperation::new(
+    svc,
+    "capture_pokemon_operation",
+)
+.request_fmt(request_fmt)
+.response_fmt(response_fmt);
+```
+Accessing the Pokédex is modeled as a restricted operation: a passcode is needed by the Pokémon trainer.
+Not to log the passcode, the code will be generated [here](https://github.com/awslabs/smithy-rs/blob/db48039065bec890ef387385773b37154b555b14/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerHttpSensitivityGenerator.kt#L58) as:
+```rust
+let request_fmt = aws_smithy_http_server::logging::sensitivity::RequestFmt::new()
+    .header(|name: &http::header::HeaderName| {
+        #[allow(unused_variables)]
+        let name = name.as_str();
+        let name_match = matches!(name, "passcode");
+        let key_suffix = None;
+        let value = name_match;
+        aws_smithy_http_server::logging::sensitivity::headers::HeaderMarker {
+            value,
+            key_suffix,
+        }
+    })
+    .label(|index: usize| matches!(index, 1));
+let response_fmt = aws_smithy_http_server::logging::sensitivity::ResponseFmt::new();
+```
+Each route is a pair, [`BoxCloneService`](https://docs.rs/tower/latest/tower/util/struct.BoxCloneService.html) wrapping the service operation (the implementation) and
+the information to consume the service operation.
+```rust
+(tower::util::BoxCloneService::new(svc), capture_pokemon_operation_request_spec)
+```
+Now the `Router` is built. `Router` is not code generated, instead lives in the aws-smithy-http-server crate; it is, whenever possible, better to write Rust code in Rust, rather than generate it to aid development.
+A `Router` is a tower `Service` that routes requests to the implemented services; hence it [implements](https://github.com/awslabs/smithy-rs/blob/db48039065bec890ef387385773b37154b555b14/rust-runtime/aws-smithy-http-server/src/routing/mod.rs#L302) `Service`
+like the other operations.
+
+To re-use code and not to overload a router with possibly unneeded operations, it also [implements](https://github.com/awslabs/smithy-rs/blob/db48039065bec890ef387385773b37154b555b14/rust-runtime/aws-smithy-http-server/src/routing/mod.rs#L146)
```

Comment:
> To re-use code and not to overload a router with possibly unneeded operations

I don't understand this phrase.

Reply:
I meant to give the reason why one would implement Layer, but I've removed this line